### DISCUSSION TITLE
More accessibility customization - take background component out of focus order

### DISCUSSION THREAD

**Discussion by ally-bryand** [gorhom/react-native-bottom-sheet#1599] [Ideas]
Hi @gorhom. 

My team is currently using your bottom sheet, but the bottom sheet currently has some built-in a11y features that don't align with our needs. Specifically, the `BottomSheetBackgroundComponent` is `accessible`, and shows up in the accessibility focus order. We need the option to set that `accessible` prop to `false` for it to not be focusable. Here is the code I'm referring to: 

```
// BottomSheetBackground.tsx

const BottomSheetBackgroundComponent = ({
  pointerEvents,
  style,
}: BottomSheetBackgroundProps) => (
  <View
    pointerEvents={pointerEvents}
    **accessible={true}**
    accessibilityRole="adjustable"
    accessibilityLabel="Bottom Sheet"
    style={[styles.container, style]}
  />
);
 ```

Would you be okay with making `accessible` modifiable through props? If so, I would be okay with attempting to do the work myself and submitting an MR.

**Comment by consuelo-sanna** [comment]
Hi guys, 
I have the same problem and I'm currently using a patch for setting the accessibility properties, there are already a few PRs for v2 and v4, I can make a PR for v5 if needed

Anyway, I've noticed that if I set the BottomSheetBackgroundComponent accessible={false}, I CAN'T close the Bottom Sheet with the voice over if I "magic tap" in the backDrop, unless the modal is far smaller then the screen.

Have you @ally-bryand the same problem?
Does someone know why setting the background to not accessible only passes the event if the bottom sheet does not cover more than 80% of the screen ??

**Reply by ally-bryand** [reply]
Hey @consuelo-sanna. We haven't run into that specific issue since we are not using `BottomSheetBackgroundComponent`. But we have run into issues where our app behaves differently based on the a11y tools being open or not. We had to resort to checking if the tools were open and conditional change our components/props based on that.

**Reply by consuelo-sanna** [reply]
Hi sorry, do you mean you are not using the `BottomSheetBackDropComponent` perhaps?
The backgroundComponent is part of the bottom sheet as far as I know, unless you set it to null or use a custom one (but based on this discussion I don't think that's the case)

**Reply by ally-bryand** [reply]
Sorry, you are right and my wording was inaccurate. We are using the `BottomSheetBackgroundComponent`, we just haven't set `accessible={false}`.

To clarify, we want the ability to do that. We just haven't forked the code or anything like that in order to modify that prop or add the ability to pass the value in as a prop

**Comment by jeff-marshall-tvo** [comment]
VoiceOver cannot reach or focus interactive elements inside BottomSheetModal content — modal boundary and handle/children accessibility scoping issue

### Environment

@gorhom/bottom-sheet: [your version]
React Native: [your version]
Platform: iOS Simulator (also worth testing on-device — noting here in case others confirm/deny device parity)
Accessibility tool: VoiceOver / Accessibility Inspector

### Summary

When a BottomSheetModal contains interactive elements (e.g. buttons) alongside a custom handleComponent, VoiceOver's swipe navigation cannot reliably reach those interactive elements. Focus instead loops between the background/backdrop and the handle's content, never entering the scrollable/content area where the interactive elements live.

### Reproduction:
```

<BottomSheetModal
  ref={sheetRef}
  snapPoints={['40%']}
  backdropComponent={renderBackdrop}
  handleComponent={renderHandle}
>
  <BottomSheetView accessibilityViewIsModal={true}>
    <Text>Some status text</Text>
    <View style={styles.buttonRow}>
      <TouchableOpacity
        onPress={onPrimaryPress}
        accessible={true}
        accessibilityRole="button"
        accessibilityLabel="Primary action"
      >
        <Text>Primary action</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={onSecondaryPress}
        accessible={true}
        accessibilityRole="button"
        accessibilityLabel="Secondary action"
      >
        <Text>Secondary action</Text>
      </TouchableOpacity>
    </View>
  </BottomSheetView>
</BottomSheetModal>

```
### Observed behavior

With VoiceOver on, swiping right from the top of the screen after the sheet opens produces this traversal order:

Backdrop/background element
Handle content (e.g. a title rendered via handleComponent)
The content BottomSheetView region as one collapsed block
Back to the backdrop — the buttons are never individually reached

### Expected behavior

Focus should enter the sheet's content once presented, exclude the backdrop and underlying screen entirely (standard modal behavior), and traverse each interactive element in order.

### What I tried, none of which resolved it:

Setting _accessibilityViewIsModal={true}_ on the outer content wrapper (both alone and combined with importantForAccessibility on Android)
Ensuring handleComponent returns a single rooted view (rather than a fragment) so it isn't a disconnected sibling branch from children
_collapsable={false}_ on wrapper views, to rule out RN's view-flattening optimization from dropping the accessibility props
Imperative focus via _AccessibilityInfo.setAccessibilityFocus(node)_ on the interactive element after the sheet's onChange reports the open index, including polling UIManager.measure until the node reports a non-zero layout frame before calling focus. The call fires without error against a valid, non-zero-sized node, but VoiceOver does not land on it.

### My working theory

handleComponent and children appear to render into separate branches under the sheet's portal, rather than one continuous accessible subtree. This seems to prevent a single accessibilityViewIsModal boundary (or Android's equivalent isolation) from correctly scoping the whole presented sheet, which may be why focus keeps resolving back out to the backdrop instead of proceeding through the sheet's content in order.

**Question for maintainers:** Is there a recommended pattern for scoping a single modal accessibility boundary around both handleComponent and children together (e.g. via containerComponent, if that's intended for this)? If not, would this be a reasonable feature request, or is there prior discussion/context on this that I should be pointing to instead of filing new?

Happy to provide a minimal repro repo if that would help triage this.
